alica_engine/src/engine/PlanBase.cpp

```diff
@@ -338,19 +338,19 @@ void PlanBase::stop()
 
     if (_mainThread != nullptr) {
         _mainThread->join();
-        delete _mainThread;
+        _mainThread.reset();
```

Comment:
Nit: not needed

Reply:
Just keep the previous behaviour of the code. And protection from error in case of calling `stop` twice in a row.